flake8_trio/visitors/helpers.py

```diff
@@ -29,7 +29,9 @@
 
     T = TypeVar("T", bound=Flake8TrioVisitor)
     T_CST = TypeVar("T_CST", bound=Flake8TrioVisitor_cst)
-    T_EITHER = TypeVar("T_EITHER", bound=Flake8TrioVisitor | Flake8TrioVisitor_cst)
+    T_EITHER = TypeVar(
+        "T_EITHER", bound=Union[Flake8TrioVisitor, Flake8TrioVisitor_cst]
+    )
```

Comment:
Why this change?

Reply:
I didn't previously specify `python_version` in `pyproject.toml` for mypy, and now that I did it didn't like the old version which wouldn't work at runtime - and mypy doesn't seem to detect that we're inside an `if TYPE_CHECKING` block.

Reply:
although I never remember which tools want `python_version` to be set to max-version-you-support vs min-version-you-support, so maybe should set it to 3.12 instead of 3.9

Reply:
mypy wants you to run separate runs of all of them in parallel x)
https://github.com/python/mypy/issues/12286